crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -947,10 +947,9 @@ impl PgManager {
             None => return Ok(None),
         };
 
-        let record = field_record_object
-            .to_rust::<Field<Domain, NameRecord>>()
-            .ok_or_else(|| Error::Internal(format!("Malformed Object {record_id}")))?
-            .value;
+        let record = NameRecord::try_from(field_record_object)?;
+
+        // TODO(manos): Implement new logic.
```

Comment:
what's the new logic? the validation around expiration?

Reply:
Yes!